python/lsst/obs/subaru/objectMasks.py

```diff
@@ -0,0 +1,132 @@
+    """
+    def __init__(self):
+        schema = afwTable.SimpleTable.makeMinimalSchema()
+
+        schema.addField("radius", 'Angle', "radius of mask")
+        self._catalog = afwTable.SimpleCatalog(schema)
+        self._catalog.table.setMetadata(dafBase.PropertyList())
+
+        self.table  = self._catalog.table
+        self.addNew = self._catalog.addNew
+
+    def __len__(self):
+        return len(self._catalog)
+
+    def __iter__(self):
+        return iter(self._catalog)
+
+    def __getitem__(self, i):
+        return self._catalog.__getitem__(i)
+
+    def __setitem__(self, i, v):
+        return self._catalog.__setitem__(i, v)
+
+    @staticmethod
+    def readFits(fileName, hdu=0, flags=0):
+        """Read a ds9 region file, returning a ObjectMaskCatalog object
+
+        This method is called "readFits" to fool the butler. The corresponding mapper entry looks like
+        brightObjectMask: {
+            template:      "deepCoadd/BrightObjectMasks/%(tract)d/BrightObjectMask-%(tract)d-%(patch)s-%(filter)s.reg"
+            python:        "lsst.obs.subaru.objectMasks.ObjectMaskCatalog"
+            persistable:   "PurePythonClass"
+            storage:       "FitsCatalogStorage"
+        }
+        and this is the only way I know to get it to read a random file type, in this case a ds9 region file
+        """
+
+        log = pexLog.getDefaultLog().createChildLog("ObjectMaskCatalog")
+
+        brightObjects = ObjectMaskCatalog()
+        checkedWcsIsFk5 = False
+
+        with open(fileName) as fd:
+            for lineNo, line in enumerate(fd.readlines(), 1):
+                line = line.rstrip()
+
+                if re.search(r"^\s*#", line):
+                    #
+                    # Parse any line of the form "# key : value" and put them into the metadata.
+                    #
+                    # As noted in HSC-1342, we expect to see at least the keys
+                    #   tract patch filter
+                    # (they will be converted to lower case, in case we switch to using FITS,
+                    # whose keys are case insensitive).  The value of these three keys will be checked,
+                    # so get them right!
+                    #
+                    mat = re.search(r"^\s*#\s*([a-zA-Z][a-zA-Z0-9_]+)\s*:\s*(.*)", line)
+                    if mat:
+                        key, value = mat.group(1).lower(), mat.group(2)
+                        if key == "tract":
+                            value = int(value)
+
+                        brightObjects.table.getMetadata().set(key, value)
+
+                line = re.sub(r"^\s*#.*", "", line)
+                if not line:
+                    continue
+
+                if re.search(r"^\s*wcs\s*;\s*fk5\s*$", line, re.IGNORECASE):
+                    checkedWcsIsFk5 = True
+                    continue
+
+                mat = re.search(r"^\s*circle(?:\s+|\s*\(\s*)"
```

Comment:
A comment describing exactly what this is supposed to match would be helpful. Indeed, for bonus points, describe in the docstring of this method exactly what the file format you're expecting to read is, providing e.g. links to the DS9 spec where appropriate.